README.md

```diff
@@ -139,6 +142,89 @@ To upgrade this resource from `4.0.6` to `4.2.7`:
    kubectl apply -f <example>.yaml --namespace <my-namespace>
    ```
 
+## Secure MongoDB Resource Connections using TLS
+
+You can configure the MongoDB Community Kubernetes Operator to use TLS certificates to encrypt traffic between:
+
+- MongoDB hosts in a replica set, and
+- Client applications and MongoDB deployments.
+
+### Prerequisites
+
+Before you secure MongoDB resource connections using TLS, you must:
+
+1. Using your own Certificate Authority (CA), create a PEM-encoded TLS certificate for each member of the Application Database’s replica set. The certificate must have one of the following:
+
+   - A wildcard `Common Name` that matches the domain name of all of the replica set members:
+
+     ```
+     *.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     ```
+   - The domain name for each of the replica set members as `Subject Alternative Names` (SAN):
+
+     ```
+     <metadata.name of the MongoDB resource>-0.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     <metadata.name of the MongoDB resource>-1.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     <metadata.name of the MongoDB resource>-2.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     ```
+
+1. Create a Kubernetes ConfigMap that contains the certificate for the CA that signed your server certificate. For a certificate file named `ca.crt`:
```

Comment:
I think it's worth clarifying here that the key within the ConfigMap needs to be called `ca.crt`. This will happen automatically when using `--from-file=ca.crt` but not if the user has their certificate stored with another file name.

This does not apply for the Secret though. `kubectl` will automatically make sure the certificate and private key end up under `tls.crt` and `tls.key` respectively, no matter the file names used.

Reply:
updated